src/collection-facets/more-facets-content.ts

```diff
@@ -0,0 +1,361 @@
+    }
+
+    if (changed.has('aggregations')) {
+      this.filterFacets();
+    }
+  }
+
+  async fetchSpecificFacets(facetAggregationKey: string) {
+    const aggregations = {
+      advancedParams: [
+        {
+          field: facetAggregationKey,
+          size: 1000000, // todo - do we want to have all the records at once?
+        },
+      ],
+    };
+
+    const params: SearchParams = {
+      query: this.fullQuery as string,
+      fields: ['identifier'],
+      aggregations,
+      rows: 1, // todo - do we want server-side pagination with offset/page/limit flag?
+    };
+
+    const results = await this.searchService?.search(params);
+    this.aggregations = results?.success?.response.aggregations as any;
+  }
+
+  async filterFacets() {
+    Object.entries(this.aggregations ?? []).forEach(([key, buckets]) => {
+      if (key === 'year_histogram') return;
+
+      this.castedBuckets = buckets['buckets'] as Bucket[];
+
+      if (this.facetKey === 'collection') {
+        // for collections, we need to asynchronously load the collection name
+        // so we use the `async-collection-name` widget and for the rest, we have a static value to use
+        const collectionIds = this.castedBuckets?.map(option => option.key);
+        const collectionIdsArray = Array.from(
+          new Set(collectionIds)
+        ) as string[];
+        this.collectionNameCache?.preloadIdentifiers(collectionIdsArray);
+      }
+    });
+
+    const { length } = Object.keys(this.castedBuckets as []);
+    this.paginationSize = Math.ceil(length / this.facetsPerPage);
+  }
+
+  private pageNumberClicked(e: CustomEvent<{ page: string }>) {
+    const page = e?.detail?.page;
+    if (page) {
+      this.pageNumber = Number(page);
+    }
+  }
+
+  private get getMoreFacetsTemplate() {
+    this.facetsLoading = false;
+
+    // render only items which will be visible as per this.facetsPerPage
+    const currentPageContent = this.castedBuckets?.slice(
+      (this.pageNumber - 1) * this.facetsPerPage,
+      this.pageNumber * this.facetsPerPage
+    );
+
+    return html`<ul class="facet-list">
+      ${currentPageContent?.map(facet => {
+        let displayText = facet.key;
+
+        if (this.facetKey === 'language') {
+          displayText =
+            this.languageCodeHandler?.getLanguageNameFromCodeString(
+              displayText as string
+            ) ?? displayText;
+        }
+
+        return html` <li class="facet-row">
+          <label class="facet-info-display" title=${facet.key}>
+            <input
+              type="checkbox"
+              class="selected-facets"
+              .value="${facet.key}"
+              data-facet="${this.facetKey}"
+              @click=${(e: Event) => {
+                this.facetClicked(e);
+              }}
+            />
+            <div class="facet-title">
+              ${this.facetKey !== 'collection'
+                ? html`${displayText}`
+                : html`<async-collection-name
```

Comment:
Q: where is this component?

Reply:
its part of https://github.com/internetarchive/iaux-collection-name-cache/tree/main/src to render cached value for collection. Its being used by jason